### PR TITLE
Add support for tuple assignment in update statements.

### LINKED ISSUE #2476: sql: support tuple assignment in UPDATE statements
```
UPDATE test SET (a, b) = (1, 2)
```

`parser.UpdateExpr` will need to be generalized to take a list of qualified names.

### PR DESCRIPTION
**Pull request by petermattis**
For example: "UPDATE test SET (a, b) = (1, 2)".

Cleaned up some of the rule types in sql.y to allow removing various
type assertions.

Fixes #2476.

### CODE REVIEW

**Comment by mberhault** [comment]
LGTM

File: sql/update.go [hunk 1]
```diff
@@ -44,7 +45,31 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
...
+		if expr.Tuple {
+			// TODO(pmattis): The distinction between Tuple and DTuple here is
+			// irritating. We'll see a DTuple if the expression was a subquery that
+			// has been evaluated. We'll see a Tuple in other cases.
+			n := 0
+			switch t := expr.Expr.(type) {
+			case parser.Tuple:
+				n = len(t)
+			case parser.DTuple:
+				n = len(t)
```
Comment (vivekmenezes) [inline sql/update.go]: I'm pretty sure you can write this as one case parser.Tuple, parser.DTuple even when they are different types.

File: sql/update.go [hunk 2]
```diff
@@ -65,11 +90,27 @@ func (p *planner) Update(n *parser.Update) (planNode, error) {
...
 	targets = append(targets, parser.StarSelectExpr())
 	for _, expr := range n.Exprs {
-		targets = append(targets, parser.SelectExpr{Expr: expr.Expr})
+		if expr.Tuple {
+			switch t := expr.Expr.(type) {
+			case parser.Tuple:
+				for _, e := range t {
+					targets = append(targets, parser.SelectExpr{Expr: e})
+				}
+			case parser.DTuple:
```
Comment (vivekmenezes) [inline sql/update.go]: same here

**Comment by vivekmenezes** [comment]
LGTM

File: sql/update.go (hunk 1 above)
Comment (petermattis) [inline sql/update.go]: If you write it as `case parser.Tuple, parser.DTuple:`, then `t` has the type `parser.Expr` for which `len(t)` is invalid.

Comment (vivekmenezes) [inline sql/update.go]: Tommyrot! :-)

**Pull request merged**